trie/stacktrie.go

```diff
@@ -95,10 +97,11 @@ func (t *StackTrie) Update(key, value []byte) error {
 	}
 	k = k[:len(k)-1] // chop the termination flag
 
-	// Track the key of first inserted entry as the left boundary.
-	if t.leftBoundary == nil {
-		t.leftBoundary = append([]byte{}, k...)
+	if t.first == nil {
+		t.first = append([]byte{}, k...)
 	}
+	t.last = append([]byte{}, k...)
```

Comment:
It would be neat to not have to copy this (since it's an extra copy for every inserted item). I think it's not needed, the `k` is freshly allocated in this scope (not external access), and even though a stNode may have a reference to it, I don't think it will ever be modified.

Reply:
Alternatively, we could have `t.last` be initialized as a e..g 32-byte slice, and then we could do `t.last = append(t.last[:0], k...)` or something to that effect

Reply:
> Personally I don't like hexToCompactInPlace optimization, because it might introduce some nasty issues which are pretty hard to debug.

I agree!